C#/Botkit/botbuilder-slack-adapter/SlackMessageTypeMiddleware.cs

```diff
@@ -14,9 +42,65 @@ public class SlackMessageTypeMiddleware : MiddlewareSet
         /// </summary>
         /// <param name="context"></param>
         /// <param name=""></param>
-        public async void OnTurn(TurnContext context, Func<object>)
+        public async void OnTurn(TurnContext context, Func<Task<object>> next)
         {
+            if (context.Activity.Type == "message" && context.Activity.ChannelData != null)
+            {
+                var adapter = context.Adapter as SlackAdapter;
+
+                string bot_user_id = await adapter.GetBotUserByTeam(context.Activity); // TODO const
```

Comment:
I noticed that the `// TODO const` comments came from the TS version. We should think about why this variable should be const or not. In case it's not necessary, remove the comment

Reply:
This was absolutely an oversight on my side. That comment shouldn't have made it into the PR. My bad